apps/server/src/modules/authentication/controllers/login.controller.ts

```diff
@@ -12,14 +16,38 @@
 	@UseGuards(AuthGuard('ldap'))
 	@HttpCode(HttpStatus.OK)
 	@Post('ldap')
-	loginLdap(@Req() req: { user: ICurrentUser }) {
-		return this.authService.generateJwt(req.user);
+	loginLdap(@CurrentUser() user: ICurrentUser) {
+		return this.authService.generateJwt(user);
 	}
 
 	@UseGuards(AuthGuard('local'))
 	@HttpCode(HttpStatus.OK)
 	@Post('local')
-	loginLocal(@Req() req: { user: ICurrentUser }) {
-		return this.authService.generateJwt(req.user);
+	loginLocal(@CurrentUser() user: ICurrentUser) {
+		return this.authService.generateJwt(user);
+	}
+
+	@UseGuards(AuthGuard('oauth'))
+	@HttpCode(HttpStatus.OK)
+	@Get('oauth/:systemId')
+	async loginOauth(
```

Comment:
There is no error handling on this controller to redirect the user to an error page or back to the login page. This has to be done, otherwise the user gets stuck on this endpoint, since he is actually being redirected here. I think this is a general problem with errors during the passport strategy.